tools/run_tests/run_tests.py

```diff
@@ -155,6 +156,8 @@ def configure(self, config, args):
                                                                        self.args.compiler)
 
   def test_specs(self):
+    if self.args.compiler is _FREE_PASS and self.args.travis:
```

Comment:
I still think it's very dangerous to just accept any compiler name and pretend the tests have passed. Imagine someone makes a typo while setting up the jenkins testing matrix (I can easily see someone typing "python36" instead of "python3.6" and thinking how great it is the tests are all green).